overcast.py

```diff
@@ -77,25 +85,34 @@ def get_episode_time_remaining_seconds(self, episode_id, episode_html):
     def get_all_podcasts(self):
         doc = self._get_html('https://overcast.fm/podcasts')
         return [
-            {'id': cell.attrib['href'].lstrip('/'),
-            'title': cell.cssselect('div.title')[0].text_content(),
-            'albumArtURI': cell.cssselect('img')[0].attrib['src'],
+            {
+                'id': cell.attrib['href'].lstrip('/'),
+                'title': cell.cssselect('div.title')[0].text_content(),
+                'albumArtURI': cell.cssselect('img')[0].attrib['src'],
             }
             for cell in doc.cssselect('a.feedcell')
             if 'href' in cell.attrib
         ]
 
-    def get_all_podcast_episodes(self, podcast_id, limit=10):
+    def get_all_podcast_episodes(self, podcast_id):
         """
         get all episodes (played or not) for a podcast.
-
-        needs to be limited to avoid hammering overcast and timing out sonos.
         """
         podcast_href = urlparse.urljoin('https://overcast.fm', podcast_id)
         doc = self._get_html(podcast_href)
+        albumArtURI = doc.cssselect('img.art')[0].attrib['src']
+        podcast_title = doc.cssselect('h2.centertext')[0].text_content()
         return [
-            self.get_episode_detail(cell.attrib['href'])
-            for cell in doc.cssselect('a.extendedepisodecell')[:limit]
+            # NOTE: If the hardcoded audio_type causes any problems, just uncomment the line below and comment out the dictionary below it.
+            # self.get_episode_detail(cell.attrib['href'])
+            {
+                'id': urlparse.urljoin('https://overcast.fm', cell.attrib['href']).lstrip('/'),
+                'title': cell.cssselect('div.titlestack div.title')[0].text_content(),
+                'audio_type': 'audio/mpeg',
+                'podcast_title': podcast_title,
+                'albumArtURI': albumArtURI,
+            }
+            for cell in doc.cssselect('a.extendedepisodecell')
```

Comment:
This means we no longer need to make 2 requests per episode, and only once queued/played are the necessary calls made! In my testing, I have found that `'audio_type'` simply needs to be present valid, but can be different from the final played file. I tested setting it to `audio/aac` even though the actual file I tested with was `audio/mpeg` and it still worked. I will leave the note for future clarity in case something ever breaks.